# Data

Home credit wants to unlock the full potential of their data using Machine Learning. They have shared the data about  

1. **Application and applicant's details**
2. **Her loan history, if any, with Home Credit**
3. **Her loan history, if any, with other institution received from Credit Bureau**
  
All this information is avaialbe in 7 files. 

1. **application_{train|test}.csv** - The main file with information about loan and loan applicants. Each row represents an unique loan and whether it was repaid or defaulted on.    <br />
  <br />
2. **previous_application.csv** - All previous applications for Home Credit loans of clients who have loans in our sample. Each row is one prevoius application.      <br />
  <br />
3. **POS_CASH_balance.csv** - Monthly balance snapshots of previous POS (point of sales) and cash loans that the applicant had with Home Credit.This table has one row for each month of history of every previous credit in Home Credit (consumer credit and cash loans) related to loans in our sample.  <br /> 
  i.e. the table has (# of loans in sample * # of relative previous credits * # of months in which we have some history observable for the previous credits) rows.    <br />
  <br />
4. **credit_card_balance.csv** - Monthly balance snapshots of previous credit cards that the applicant has with Home Credit. This table has one row for each month of history of every previous credit in Home Credit (consumer credit and cash loans) related to loans in our sample.  <br />
i.e. the table has (# of loans in sample * # of relative previous credit cards * # of months where we have some history observable for the previous credit card) rows.      <br />
  <br />
5. **installments_payments.csv** - Repayment history for the previously disbursed credits in Home Credit related to the loans in our sample. There is a) one row for every payment that was made plus b) one row each for missed payment. One row is equivalent to one payment of one installment OR one installment corresponding to one payment of one previous Home Credit credit related to loans in our sample.  <br />
  <br />
6. **bureau.csv** - All client's previous credits provided by other financial institutions that were reported to Credit Bureau (for clients who have a loan in our sample). For every loan in our sample, there are as many rows as number of credits the client had in Credit Bureau before the application date.  <br />
  <br />
  
7. **bureau_balance.csv** - Monthly balances of previous credits in Credit Bureau. This table has one row for each month of history of every previous credit reported to Credit Bureau – i.e the table has (#loans in sample * # of relative previous credits * # of months where we have some history observable for the previous credits) rows.  <br />

We also have data dictionary with the decription of each variable. Let us expore it to get a broad view of data.

In [1]:
import pandas as pd
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_colwidth', None)

dictionary = pd.read_csv('../docs/data dictionary/HomeCredit_columns_description.csv')

In [2]:
dictionary.head()

Unnamed: 0                         Table                 Row  \
0           1  application_{train|test}.csv          SK_ID_CURR   
1           2  application_{train|test}.csv              TARGET   
2           5  application_{train|test}.csv  NAME_CONTRACT_TYPE   
3           6  application_{train|test}.csv         CODE_GENDER   
4           7  application_{train|test}.csv        FLAG_OWN_CAR   

                                                                                                                                                                                   Description  \
0                                                                                                                                                                     ID of loan in our sample   
1  Target variable (1 - client with payment difficulties: he/she had late payment more than X days on at least one of the first Y installments of the loan in our sample, 0 - all other cases)   
2                                                                                                                                                  Identification if loan is cash or revolving   
3                                                                                                                                                                         Gender of the client   
4                                                                                                                                                                Flag if the client owns a car   

  Special  
0     NaN  
1     NaN  
2     NaN  
3     NaN  
4     NaN

In [3]:
dictionary.drop('Unnamed: 0',axis=1,inplace=True)
dictionary.shape

(219, 4)

In [4]:
dictionary.groupby('Table').Row.agg('count')

Table
POS_CASH_balance.csv              8
application_{train|test}.csv    122
bureau.csv                       17
bureau_balance.csv                3
credit_card_balance.csv          23
installments_payments.csv         8
previous_application.csv         38
Name: Row, dtype: int64

We have 219  variables in total. Majority of the variables are from current application & previous application.

In [5]:
dictionary

Table                           Row  \
0    application_{train|test}.csv                    SK_ID_CURR   
1    application_{train|test}.csv                        TARGET   
2    application_{train|test}.csv            NAME_CONTRACT_TYPE   
3    application_{train|test}.csv                   CODE_GENDER   
4    application_{train|test}.csv                  FLAG_OWN_CAR   
5    application_{train|test}.csv               FLAG_OWN_REALTY   
6    application_{train|test}.csv                  CNT_CHILDREN   
7    application_{train|test}.csv              AMT_INCOME_TOTAL   
8    application_{train|test}.csv                    AMT_CREDIT   
9    application_{train|test}.csv                   AMT_ANNUITY   
10   application_{train|test}.csv               AMT_GOODS_PRICE   
11   application_{train|test}.csv               NAME_TYPE_SUITE   
12   application_{train|test}.csv              NAME_INCOME_TYPE   
13   application_{train|test}.csv           NAME_EDUCATION_TYPE   
14   application_{train|test}.csv            NAME_FAMILY_STATUS   
15   application_{train|test}.csv             NAME_HOUSING_TYPE   
16   application_{train|test}.csv    REGION_POPULATION_RELATIVE   
17   application_{train|test}.csv                    DAYS_BIRTH   
18   application_{train|test}.csv                 DAYS_EMPLOYED   
19   application_{train|test}.csv             DAYS_REGISTRATION   
20   application_{train|test}.csv               DAYS_ID_PUBLISH   
21   application_{train|test}.csv                   OWN_CAR_AGE   
22   application_{train|test}.csv                    FLAG_MOBIL   
23   application_{train|test}.csv                FLAG_EMP_PHONE   
24   application_{train|test}.csv               FLAG_WORK_PHONE   
25   application_{train|test}.csv              FLAG_CONT_MOBILE   
26   application_{train|test}.csv                    FLAG_PHONE   
27   application_{train|test}.csv                    FLAG_EMAIL   
28   application_{train|test}.csv               OCCUPATION_TYPE   
29   application_{train|test}.csv               CNT_FAM_MEMBERS   
30   application_{train|test}.csv          REGION_RATING_CLIENT   
31   application_{train|test}.csv   REGION_RATING_CLIENT_W_CITY   
32   application_{train|test}.csv    WEEKDAY_APPR_PROCESS_START   
33   application_{train|test}.csv       HOUR_APPR_PROCESS_START   
34   application_{train|test}.csv    REG_REGION_NOT_LIVE_REGION   
35   application_{train|test}.csv    REG_REGION_NOT_WORK_REGION   
36   application_{train|test}.csv   LIVE_REGION_NOT_WORK_REGION   
37   application_{train|test}.csv        REG_CITY_NOT_LIVE_CITY   
38   application_{train|test}.csv        REG_CITY_NOT_WORK_CITY   
39   application_{train|test}.csv       LIVE_CITY_NOT_WORK_CITY   
40   application_{train|test}.csv             ORGANIZATION_TYPE   
41   application_{train|test}.csv                  EXT_SOURCE_1   
42   application_{train|test}.csv                  EXT_SOURCE_2   
43   application_{train|test}.csv                  EXT_SOURCE_3   
44   application_{train|test}.csv                APARTMENTS_AVG   
45   application_{train|test}.csv              BASEMENTAREA_AVG   
46   application_{train|test}.csv   YEARS_BEGINEXPLUATATION_AVG   
47   application_{train|test}.csv               YEARS_BUILD_AVG   
48   application_{train|test}.csv                COMMONAREA_AVG   
49   application_{train|test}.csv                 ELEVATORS_AVG   
50   application_{train|test}.csv                 ENTRANCES_AVG   
51   application_{train|test}.csv                 FLOORSMAX_AVG   
52   application_{train|test}.csv                 FLOORSMIN_AVG   
53   application_{train|test}.csv                  LANDAREA_AVG   
54   application_{train|test}.csv          LIVINGAPARTMENTS_AVG   
55   application_{train|test}.csv                LIVINGAREA_AVG   
56   application_{train|test}.csv       NONLIVINGAPARTMENTS_AVG   
57   application_{train|test}.csv             NONLIVINGAREA_AVG   
58   application_{train|test}.csv               APARTMENTS_MODE   
59   app

source : https://www.kaggle.com/c/home-credit-default-risk/data


A kaggler has kindly provided some insights about domain knowledge at  
https://www.kaggle.com/c/home-credit-default-risk/discussion/63032